README.md

```diff
@@ -1,95 +1,209 @@
+| 1B | 2.5 GB | 1.25GB | 0.75GB |
+| 3B | 6.5 GB | 3.2GB | 1.75GB |
+
+> [!NOTE]
+> These are estimated values and may vary based on specific
+implementation details and optimizations.
+
+Working with the capable Llama 3.1 8B models:
+
+* [Run Llama 3.1 8B in 4-bits with bitsandbytes](./4bit_bnb.ipynb)
+* [Run Llama 3.1 8B in 8-bits with bitsandbytes](./8bit_bnb.ipynb)
+* [Run Llama 3.1 8B with AWQ & fused ops](./awq.ipynb)
+
+Working on the 🐘 big Llama 3.1 405B model:
+
 * [Run Llama 3.1 405B FP8](./fp8-405B.ipynb)
 * [Run Llama 3.1 405B quantized to INT4 with AWQ](./awq_generation.py)
 * [Run Llama 3.1 405B quantized to INT4 with GPTQ](./gptq_generation.py)
-* [Run assisted decoding with Llama 405B and Llama 8B](./assisted_decoding.py)
+
+## Model Fine Tuning:
+
+It is often not enough to run inference on the model, you would also
+need to fine tune the model on some custom dataset. We have got you
+covered.
+
+Fine tune models on your custom dataset:
+* [Fine tune Llama 3.2 Vision on a custom dataset](./Llama-Vision%20FT.ipynb)
+* [How to fine-tune Llama 3.1 8B on consumer GPU with PEFT and QLoRA with bitsandbytes](./peft_finetuning.py)
+* [Execute a distributed fine tuning job for the Llama 3.1 405B model on a SLURM-managed computing cluster](./qlora_405B.slurm)
+
+## Assisted Decoding Techniques
+
+Do you want to use the smaller Llama 3.2 models to speedup text generation
+of bigger models? Here we talk about assisted decoding (speculative decoding), which gives you ~2x speedups for text generation on Llama 3.1 70B (with greedy decoding).
+
+* [Run assisted decoding with 🐘 Llama 3.1 70B and 🤏 Llama 3.2 3B](./assisted_decoding_70B_3B.ipynb)
+* [Run assisted decoding with Llama 3.1 8B and Llama 3.2 1B](./assisted_decoding_8B_1B.ipynb)
+
+## Performance Optimization
+
+Let us optimize performace shall we?
+
 * [Accelerate your inference using torch.compile](./torch_compile.py)
 * [Accelerate your inference using torch.compile and 4-bit quantization with torchao](./torch_compile_with_torchao.ipynb)
-* Execute some Llama-generated Python code
-* Use tools with Llama!
+* [Quantize KV Cache to lower memory requirements](./quantized_cache.py)
+* [How to reuse prompts with dynamic caching](./prompt_reuse.py)
 
 ## API inference
 
 Are these models too large for you to run at home? Would you like to experiment with Llama 405B? Try out the following examples!
 
 * [Use the Inference API for PRO users](./inference-api.ipynb)
-* Use a dedicated Inference Endpoint
 
 ## Llama Guard and Prompt Guard
 
 In addition to the generative models, Meta released two new models: Llama Guard 3 and Prompt Guard. Prompt Guard is a small classifier that detects jailbreaks and prompt injections. Llama Guard 3 is a safeguard model that can classify LLM inputs and generations. Learn how to use them as done in the following notebooks:
 
 * [Detecting jailbreaks and prompt injection with Prompt Guard](./prompt_guard.ipynb)
-* Using Llama Guard for Guardrailing
```

Comment:
Let's open issues for the points for which we never got to add a notebook

Reply:
Will do!